jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -723,6 +736,8 @@ public void add(HttpHeader header, HttpHeaderValue value)
      */
     public void add(HttpHeader header, String value)
     {
+        Objects.requireNonNull(header, "header");
+
         if (value == null)
             throw new IllegalArgumentException("null value");
```

Comment:
We are no-op all the other cases, most notably `add(String, String)` above, so why throw here?

Reply:
Because it will have servlet API impacts.  There is a separate issue that I thought you and and @lachlan-roberts  were working on to normalise the API in 10 to either always noop or always ISE (I can't remember which).  Either way, I don't want to change behaviour in this PR other than to avoid noops and later NPEs (eg from null names or null fields).